src/main/java/ch/vorburger/fswatch/DirectoryWatcherImpl.java

```diff
@@ -101,20 +119,28 @@ protected DirectoryWatcherImpl(boolean watchSubDirectories, final Path watchBase
                         }
                     }
 
-                    if (kind == StandardWatchEventKinds.ENTRY_MODIFY || kind == StandardWatchEventKinds.ENTRY_DELETE) {
-                        // To reduce notifications, only call the Listener on Modify and Delete but not Create,
-                        // because (on Linux at least..) every ENTRY_CREATE from new file
-                        // is followed by an ENTRY_MODIFY anyway.
+                   
                         try {
                             ChangeKind ourKind = kind == StandardWatchEventKinds.ENTRY_MODIFY ? ChangeKind.MODIFIED : ChangeKind.DELETED;
-                            listener.onChange(absolutePath, ourKind);
+                            //Cannot use switch here as "kind" is not one of the supported types for switch.
+                            if (kind ==  StandardWatchEventKinds.ENTRY_CREATE) {
+                            	ourKind = ChangeKind.CREATED;
+                            }else if (kind ==  StandardWatchEventKinds.ENTRY_MODIFY) {
+                            	ourKind = ChangeKind.MODIFIED;
+                            }else {
+                            	ourKind = ChangeKind.DELETED;
```

Comment:
should you not check `else if (kind == StandardWatchEventKinds.ENTRY_MODIFY)` here? Just wondering!

Reply:
You mean check for ENTRY_DELETE, instead of assuming it is going to be ENTRY_DELETE ?